[BUG]Panic in AlluxioRuntime reconciler due to reflect.Value.IsNil call on struct Value

**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

While attempting to reconcile AlluxioRuntime, a panic error occurred. The error message is as follows:

```golang
2024-05-28T14:28:45.515+0800 INFO controller/controller.go:118 Observed a panic in reconciler: reflect: call of reflect.Value.IsNil on struct Value {"controller": "alluxioruntime", "controllerGroup": "[data.fluid.io](http://data.fluid.io/)", "controllerKind": "AlluxioRuntime", "AlluxioRuntime": {"name":"serverless-data","namespace":"default"}, "namespace": "default", "name": "serverless-data", "reconcileID": "c389b01e-17c2-4499-854e-1bc60d6ecea4"}
```

This error was caused by attempting to call reflect.Value.IsNil() on a struct Value, which is not allowed in Go.

The full log is as below:

```
2024-05-28T14:28:45.515+0800	INFO	controller/controller.go:118	Observed a panic in reconciler: reflect: call of reflect.Value.IsNil on struct Value	{"controller": "alluxioruntime", "controllerGroup": "data.fluid.io", "controllerKind": "AlluxioRuntime", "AlluxioRuntime": {"name":"serverless-data","namespace":"default"}, "namespace": "default", "name": "serverless-data", "reconcileID": "c389b01e-17c2-4499-854e-1bc60d6ecea4"}
panic: reflect: call of reflect.Value.IsNil on struct Value [recovered]
	panic: reflect: call of reflect.Value.IsNil on struct Value

goroutine 144 [running]:
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:119 +0x30a
panic({0x24127a0?, 0xc000187b60?})
	/usr/local/go/src/runtime/panic.go:920 +0x290
reflect.Value.IsNil({0x266d120, 0xc000001380, 0x99})
	/usr/local/go/src/reflect/value.go:1568 +0x172
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap.kubeObjectWrapper.MarshalLogObject({{0x7f4470180e78, 0xc000001380}}, {0x287a6c8, 0xc000886740})
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap/kube_helpers.go:69 +0xb3
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*jsonEncoder).AppendObject(0xc000886740, {0x7f4470180ea0, 0xc0007b3d00})
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/json_encoder.go:229 +0x8e
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*jsonEncoder).AddObject(0xc000886740, {0x268a5de, 0x11}, {0x7f4470180ea0, 0xc0007b3d00})
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/json_encoder.go:111 +0x59
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.Field.AddTo({{0x268a5de, 0x11}, 0x2, 0x0, {0x0, 0x0}, {0x249c460, 0xc0007b3d00}}, {0x287a6c8, 0xc000886740})
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/field.go:121 +0x1b1
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.addFields({0x287a6c8, 0xc000886740}, {0xc000217600, 0x2, 0x2})
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/field.go:210 +0x12a
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.consoleEncoder.writeContext({0xc000886640}, 0xc0002fa360, {0xc000217600, 0x2, 0x2})
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/console_encoder.go:141 +0x113
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.consoleEncoder.EncodeEntry({0xc000886640}, {0x2, {0xc18d7c475eb3a4b6, 0x3dbe898a, 0x36a2fa0}, {0xc0001a54c0, 0x19}, {0x2693eb0, 0x17}, {0x1, ...}, ...}, ...)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/console_encoder.go:119 +0x4f3
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap.(*KubeAwareEncoder).EncodeEntry(0xc000187890, {0x2, {0xc18d7c475eb3a4b6, 0x3dbe898a, 0x36a2fa0}, {0xc0001a54c0, 0x19}, {0x2693eb0, 0x17}, {0x1, ...}, ...}, ...)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/log/zap/kube_helpers.go:132 +0x6f9
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*ioCore).Write(0xc00085ac00, {0x2, {0xc18d7c475eb3a4b6, 0x3dbe898a, 0x36a2fa0}, {0xc0001a54c0, 0x19}, {0x2693eb0, 0x17}, {0x1, ...}, ...}, ...)
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/core.go:95 +0xc3
github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0008044e0, {0xc000217600, 0x2, 0x2})
	/go/src/github.com/fluid-cloudnative/fluid/vendor/go.uber.org/zap/zapcore/entry.go:255 +0x344
github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/zapr.(*zapLogger).Error(0xc00085abd0, {0x2856a60, 0xc0000c5c20}, {0x2693eb0, 0x17}, {0xc0002fa540, 0x2, 0x2})
	/go/src/github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/zapr/zapr.go:216 +0x245
github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/logr.Logger.Error({{0x2872a30, 0xc00085abd0}, 0x0}, {0x2856a60, 0xc0000c5c20}, {0x2693eb0, 0x17}, {0xc0002fa540, 0x2, 0x2})
	/go/src/github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/logr/logr.go:305 +0x14a
github.com/fluid-cloudnative/fluid/pkg/controllers.(*RuntimeReconciler).AddFinalizerAndRequeue(0xc0001ec000, {{0x2870578, 0xc00085aba0}, {{0xc000797156, 0x7}, {0xc000797160, 0xf}}, {0x26811ce, 0xa}, 0xc000160a00, ...}, ...)
	/go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/runtime_controller.go:298 +0x5d2
github.com/fluid-cloudnative/fluid/pkg/controllers.(*RuntimeReconciler).ReconcileInternal(0xc0001ec000, {{0x2870578, 0xc00085aba0}, {{0xc000797156, 0x7}, {0xc000797160, 0xf}}, {0x26811ce, 0xa}, 0xc000160a00, ...})
	/go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/runtime_controller.go:153 +0x1075
github.com/fluid-cloudnative/fluid/pkg/controllers/v1alpha1/alluxio.(*RuntimeReconciler).Reconcile(0xc000460360, {0x2870578, 0xc00085aba0}, {{{0xc000797156, 0x7}, {0xc000797160, 0xf}}})
	/go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/v1alpha1/alluxio/alluxio_runtime_controller.go:106 +0xbda
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc0002ec000, {0x2870578, 0xc00085aba0}, {{{0xc000797156, 0x7}, {0xc000797160, 0xf}}})
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:122 +0x1be
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0002ec000, {0x2870578, 0xc00085aba0}, {0x24a33a0, 0xc00004e6a0})
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:323 +0x465
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0002ec000, {0x28705b0, 0xc00030c6e0})
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:274 +0x3fc
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:235 +0xdf
created by github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2 in goroutine 62
	/go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:231 +0x9a5
```



**What you expect to happen:**

**How to reproduce it**

**Additional Information**

--- a/pkg/controllers/runtime_controller.go
+++ b/pkg/controllers/runtime_controller.go
@@ -295,7 +295,7 @@ func (r *RuntimeReconciler) AddFinalizerAndRequeue(ctx cruntime.ReconcileRequest
 	prevGeneration := objectMeta.GetGeneration()
 	objectMeta.SetFinalizers(append(objectMeta.GetFinalizers(), finalizerName))
 	if err := r.Update(ctx, ctx.Runtime); err != nil {
-		ctx.Log.Error(err, "Failed to add finalizer", "StatusUpdateError", ctx)
+		ctx.Log.Error(err, "Failed to add finalizer", "Runtime", ctx.Runtime)
 		return utils.RequeueIfError(err)
 	}
 	// controllerutil.AddFinalizer(ctx.Runtime, finalizer)

--- a/pkg/controllers/runtime_controller.go
+++ b/pkg/controllers/runtime_controller.go
@@ -298,10 +298,9 @@ func (r *RuntimeReconciler) AddFinalizerAndRequeue(ctx cruntime.ReconcileRequest
 		ctx.Log.Error(err, "Failed to add finalizer", "Runtime", ctx.Runtime)
 		return utils.RequeueIfError(err)
 	}
-	ctx.Log.Info("Succeed to add finalizer","Runtime", ctx.Runtime)
 	// controllerutil.AddFinalizer(ctx.Runtime, finalizer)
 	currentGeneration := objectMeta.GetGeneration()
-	ctx.Log.Info("RequeueImmediatelyUnlessGenerationChanged", "prevGeneration", prevGeneration,
+	ctx.Log.V(1).Info("RequeueImmediatelyUnlessGenerationChanged", "prevGeneration", prevGeneration,
 		"currentGeneration", currentGeneration)
 
 	return utils.RequeueImmediatelyUnlessGenerationChanged(prevGeneration, currentGeneration)

--- a/pkg/controllers/runtime_controller.go
+++ b/pkg/controllers/runtime_controller.go
@@ -298,6 +298,7 @@ func (r *RuntimeReconciler) AddFinalizerAndRequeue(ctx cruntime.ReconcileRequest
 		ctx.Log.Error(err, "Failed to add finalizer", "Runtime", ctx.Runtime)
 		return utils.RequeueIfError(err)
 	}
+	ctx.Log.Info("Succeed to add finalizer","Runtime", ctx.Runtime)
 	// controllerutil.AddFinalizer(ctx.Runtime, finalizer)
 	currentGeneration := objectMeta.GetGeneration()
 	ctx.Log.Info("RequeueImmediatelyUnlessGenerationChanged", "prevGeneration", prevGeneration,

